### PR TITLE
admin: describe traffic analysis with Wireshark

### PR DESCRIPTION
**Pull request by Totktonada**
This recipe uses the Wireshark plugin https://github.com/tarantool/tarantool-dissector created by Mikhail Galanin (@negram).

### CODE REVIEW

**Comment by Totktonada** [comment]
Friendly built by CI: https://docs.d.tarantool.io/en/doc/inspect-traffic-with-wireshark/book/admin/server_introspection/#inspect-traffic (my gratitude to @NickVolynkin).

image

I'm not very familiar with processes around the documentation repository. Is something else expected from me at this point?

File: doc/book/admin/server_introspection.rst [hunk 1]
```diff
@@ -169,6 +169,49 @@ But, to force Tarantool to release memory, you can
 call :doc:`box.snapshot() </reference/reference_lua/box_snapshot>`, stop the server instance,
 and restart it.
 
+.. _admin-inspect_traffic:
+
+--------------------------------------------------------------------------------
+Inspect traffic
+--------------------------------------------------------------------------------
```
Comment (patiencedaur) [inline doc/book/admin/server_introspection.rst]: If possible, please bring *all* header formatting in accordance with our style guide: https://www.tarantool.io/en/doc/latest/contributing/docs/markup/headings/

I. e.
```
Inspect traffic
---------------
```
not
```
---------------------------
Inspect traffic
---------------------------
```

File: doc/book/admin/server_introspection.rst [hunk 2]
```diff
@@ -169,6 +169,49 @@ But, to force Tarantool to release memory, you can
...
+
+Inspecting the binary traffic is the boring task. We offer a
+`Wireshark plugin <https://github.com/tarantool/tarantool-dissector>`_ to
+simplify analysis of Tarantool's traffic.
+
+Please, consider several simple steps to enable the plugin.
```
Comment (patiencedaur) [inline doc/book/admin/server_introspection.rst]:
```suggestion
Inspecting binary traffic is a boring task. We offer a
`Wireshark plugin <https://github.com/tarantool/tarantool-dissector>`_ to
simplify the analysis of Tarantool's traffic.

To enable the plugin, follow the steps below.
```
Let the user decide just how simple that is. We did our job by describing the process thoroughly.

File: doc/book/admin/server_introspection.rst [hunk 3]
```diff
@@ -169,6 +169,49 @@ But, to force Tarantool to release memory, you can
...
+
+    mkdir -p ~/.local/lib/wireshark/plugins
+    cd ~/.local/lib/wireshark/plugins
+    ln -s /path/to/tarantool-dissector/MessagePack.lua ./
+    ln -s /path/to/tarantool-dissector/tarantool.dissector.lua ./
+
+(Please, refer to chapter
+`Plugin folders <https://www.wireshark.org/docs/wsug_html_chunked/ChPluginFolders.html>`_
+in the Wireshark documentation for location of the plugin directory on macOS and
+Windows.)
```
Comment (patiencedaur) [inline doc/book/admin/server_introspection.rst]:
```suggestion
(For the location of the plugin directory on macOS and
Windows, please refer to the
`Plugin folders <https://www.wireshark.org/docs/wsug_html_chunked/ChPluginFolders.html>`_ chapter
in the Wireshark documentation.)
```
Here are some syntax-related remarks:
- "Please" with a comma is a desperate "please" (like someone is making the same mistake for far too long). "Please" without a comma is just polite.
- I changed the sentence structure to enhance the [topic, not comment](https://en.wikipedia.org/wiki/Topic_and_comment).

File: doc/book/admin/server_introspection.rst [hunk 4]
```diff
@@ -169,6 +169,49 @@ But, to force Tarantool to release memory, you can
...
+
+Run Wireshark GUI and ensure that the plugins are loaded:
```
Comment (patiencedaur) [inline doc/book/admin/server_introspection.rst]:
```suggestion
Run the Wireshark GUI and ensure that the plugins are loaded:
```

File: doc/book/admin/server_introspection.rst [hunk 5]
```diff
@@ -169,6 +169,49 @@ But, to force Tarantool to release memory, you can
...
+
+* Open ``Help`` -> ``About Wireshark`` -> ``Plugins``.
```
Comment (patiencedaur) [inline doc/book/admin/server_introspection.rst]:
```suggestion
* Open :guilabel:`Help` > :guilabel:`About Wireshark` > :guilabel:`Plugins`.
```
Consider using the `:guilabel:` directive to indicate GUI elements, as mentioned in our [style guide](https://www.tarantool.io/en/doc/latest/contributing/docs/markup/gui/).

File: doc/book/admin/server_introspection.rst [hunk 6]
```diff
@@ -169,6 +169,49 @@ But, to force Tarantool to release memory, you can
...
+* Find ``MessagePack.lua`` and ``tarantool.dissector.lua`` in the list.
+
+Now you can inspect Tarantool incoming and outcoming packets with user-friendly
```
Comment (patiencedaur) [inline doc/book/admin/server_introspection.rst]:
```suggestion
Now you can inspect incoming and outgoing Tarantool packets with user-friendly
```

**Review by patiencedaur — CHANGES_REQUESTED** [review]
Thanks for the patch! Here are some language-related comments.

File: doc/book/admin/server_introspection.rst (hunk 1 above)
Comment (Totktonada) [reply doc/book/admin/server_introspection.rst]: No problem. Added as a separate commit before the main change. I would appreciate if this pull request will go to `latest` branch without squashing to keep the history easy to follow.

File: doc/book/admin/server_introspection.rst (hunk 2 above)
Comment (Totktonada) [reply doc/book/admin/server_introspection.rst]: Sure, it looks better now. Updated.

File: doc/book/admin/server_introspection.rst (hunk 3 above)
Comment (Totktonada) [reply doc/book/admin/server_introspection.rst]: I agree. Updated.

File: doc/book/admin/server_introspection.rst (hunk 4 above)
Comment (Totktonada) [reply doc/book/admin/server_introspection.rst]: Updated.

File: doc/book/admin/server_introspection.rst (hunk 5 above)
Comment (Totktonada) [reply doc/book/admin/server_introspection.rst]: Missed it, thanks! Updated.

File: doc/book/admin/server_introspection.rst (hunk 6 above)
Comment (Totktonada) [reply doc/book/admin/server_introspection.rst]: Updated.

Comment (Totktonada) [reply doc/book/admin/server_introspection.rst]: Missed outcoming -> outgoing change at first glance, updated it now as well.

**Comment by Totktonada** [comment]
@patiencedaur Thank you for detailed comments! I updated the patchset.

**Review by patiencedaur — APPROVED** [review]
Thanks for the patch!

File: doc/book/admin/server_introspection.rst (hunk 1 above)
Comment (patiencedaur) [reply doc/book/admin/server_introspection.rst]: Unfortunately, only "squash and merge" is enabled for this repo.

**Pull request merged**